handlers/handlers.go

```diff
@@ -4,46 +4,57 @@ import (
     "net/http"
     "encoding/json"
     "fmt"
-    "log"
+    "sync"
+    "github.com/julienschmidt/httprouter"
     "github.com/skrzepto/IPRO-VAULT-SERVER/models"
-    _ "github.com/jinzhu/gorm/dialects/sqlite"
-    "github.com/jinzhu/gorm"
 )
 
 type Impl struct {
-    DB *gorm.DB
+  sd_map map[string]models.SensorData
+  mu sync.Mutex
 }
 
-func (i *Impl) InitDB() {
-    var err error
-    i.DB, err = gorm.Open("sqlite3", "ipro-vault.db")
+func InitGlobal() *Impl {
+    var i Impl
+    i.sd_map = make(map[string]models.SensorData)
+    return &i
+}
+
+
+func (i *Impl) POST_SensorData_ID(rw http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+    decoder := json.NewDecoder(req.Body)
+    var sd models.SensorData
+    err := decoder.Decode(&sd)
     if err != nil {
-        log.Fatalf("Got error when connect database, the error is '%v'", err)
+        panic(err)
     }
-    i.DB.LogMode(true)
+    fmt.Printf("decoded to %#v\n", sd)
+    //i.DB.Create(&t) change this to array
+    sensor_id := ps.ByName("sensor_id")
+    i.mu.Lock()
+    i.sd_map[sensor_id] = sd
+    i.mu.Unlock()
 }
 
-func (i *Impl) InitSchema() {
-    i.DB.AutoMigrate(&models.SensorData{})
+func (i *Impl) GET_SensorData_ID(rw http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+    sensor_id := ps.ByName("sensor_id")
+    i.mu.Lock()
+    sd := i.sd_map[sensor_id]
+    i.mu.Unlock()
+
+    js, err := json.Marshal(sd)
+    if err != nil {
+      http.Error(rw, err.Error(), http.StatusInternalServerError)
```

Comment:
yeah that's why its good to have it running continuously in the background or wrap it around a git hook so it happens before you commit :-)